rai/node/node.cpp

```diff
@@ -3065,6 +3077,46 @@ bool rai::handle_node_options (boost::program_options::variables_map & vm)
 			result = true;
 		}
 	}
+	else if (vm.count ("vacuum") > 0)
+	{
+		try
+		{
+			boost::filesystem::path data_path;
+			if (vm.count ("data_path"))
+			{
+				data_path = boost::filesystem::path (vm["data_path"].as <std::string> ());
+			}
+			else
+			{
+				data_path = rai::working_path ();
+			}
+
+			std::cout << "Vacuuming database copy in " << data_path << std::endl;
+			std::cout << "This may take a while..."  << std::endl;
+
+			// Scope the node so the mdb environment gets cleaned up properly before
+			// the original file is replaced with the vacuumed file.
+			{
+				inactive_node node (data_path);
+				node.node->copy_with_compaction ();
+			}
+
+			std::cout << "Finalizing" << std::endl;
+
+			// Note that these throw on failure
+			boost::filesystem::remove (data_path / "data.ldb");
```

Comment:
Maybe rename this to backup.ldb instead of removing, and remove backup.ldb?

Reply:
Could you elaborate? The last step after mdb's compaction copy is to replace the original with the vacuumed copy.

Reply:
It would be a nice bonus to retain the previous data.ldb for at least one more run, in case something bad happens during the vacuum.

Reply:
something like:

generate vacuum.ldb
delete backup.ldb
rename data.ldb backup.ldb
rename vacuum.ldb data.ldb